main_code/Case_1/warehouse_reminder.py

```diff
@@ -0,0 +1,282 @@
+
+###################
+# Zac McCullough
+# mccul157@msu.edu
+# 1/15
+###################
+
+
+###################
+# imports
+###################
+
+from typing import Tuple, List, Any
+import pandas as pd
+import datetime
+from dateutil import parser
+import smtplib
+from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+import os
+import logging
+
+###################
+# global vars
+###################
+
+SPREADSHEET_FILE_LOCATION = 'ex_1.xlsx'
+SEND_EMAILS_FROM = 'mccul157@msu.edu'
+DEBUG_VAR = True
+
+
+###################
+# classes
+###################
+
+
+class Part(object):
+
+    def __init__(self, part_id, date, amount, open, contact, from_email, server) -> None:
+        self.id = part_id
+        self.date = self.convert_to_date_obj(date)
+        self.amount = amount
+        self.open = open
+        self.email = contact
+        self.from_email = from_email
+        self.server = server
+
+    def __repr__(self):
+        return('ID: {id}, Due: {date}, Recieved: {amount}, '
+              'In-Shipping: {open}, Contact: {email}, Send From: {from_email}'.format(
+               id=self.id, date=self.date, amount=self.amount, open=self.open, email=self.email,
+               from_email=self.from_email
+        ))
+
+    def check_if_action_required(self) -> bool:
+        """
+        Determines if action is required based on date and value of open and amount
+        :return: bool
+        """
+
+        current_time = datetime.datetime.today()
+
+        # we store our date as an object called date, but datetime uses objects called datetime
+        # so we use .combine our date object with the minimal time interval to get the right
+        # kind of object
+        if datetime.datetime.combine(self.date,
+                                     datetime.datetime.min.time()) - datetime.timedelta(days=5) <= current_time:
+            return True
+        return False
+
+    def send_email(self) -> None:
```

Comment:
I know what you are trying to do, I just don't know actually the code will work or not. I will just approved it.